### PR TITLE
added backend and frontend support for users rights

### PR DESCRIPTION
**Pull request by mihai-danilescu**

### CODE REVIEW

File: src/raccoon/controllers/base.py [hunk 1]
```diff
@@ -50,7 +50,40 @@ def get(cls, request, pk=None, *args, **kwargs):
...
+            try:
+                rights.append(Right.objects.get(pk=right_id))
+            except:
+                pass
+
+        request.send(cls.filter_response(response, rights))
+
+    @classmethod
+    def check_rights(cls, object, rights):
+        return object
```
Comment (vladmunteanu) [inline src/raccoon/controllers/base.py]: Object is a reserved keyword

File: src/raccoon/controllers/base.py [hunk 2]
```diff
@@ -50,7 +50,40 @@ def get(cls, request, pk=None, *args, **kwargs):
...
+    def filter_response(cls, response, rights):
+        if not rights:
+            return response
+
+        # treat rights if there is only one object in the response
+        if type(response) is dict:
+            if cls.check_rights(response, rights):
+                return response
+            else:
+                ReplyError(404)
```
Comment (vladmunteanu) [inline src/raccoon/controllers/base.py]: Missing raise

File: web/js/components/settings/RightForm.react.js [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+    }
+
+    _onChange() {
+        let state = getLocalState();
+        state.right = this.state.right;
+        this.setState(state);
+    }
+
+    onFormChange(name, event) {
+        this.state.right[name] = event.target.value;
```
Comment (vladmunteanu) [inline web/js/components/settings/RightForm.react.js]: This mutates the state object directly

File: src/raccoon/controllers/base.py (hunk 1 above)
Comment (mihai-danilescu) [reply src/raccoon/controllers/base.py]: Replaced with 'entity'

File: src/raccoon/controllers/base.py (hunk 2 above)
Comment (mihai-danilescu) [reply src/raccoon/controllers/base.py]: Added raise

File: web/js/components/settings/RightForm.react.js (hunk 3 above)
Comment (mihai-danilescu) [reply web/js/components/settings/RightForm.react.js]: Used a local var instead of the actual state object. Should be cleaner.

**Pull request merged**